components/scg/src/scg.js

```diff
@@ -654,27 +707,30 @@ SCg.Editor.prototype = {
      * Function that process selection changes in scene
      * It updated UI to current selection
      */
-    onSelectionChanged: function () {
-        if (this.canEdit) {
-            this.hideTool(this.toolChangeIdtf());
-            this.hideTool(this.toolSetContent());
-            this.hideTool(this.toolChangeType());
-            this.hideTool(this.toolDelete());
-            if (this.scene.selected_objects.length > 1) {
-                if (this.scene.isSelectedObjectAllArcsOrAllNodes() && !this.scene.isSelectedObjectAllHaveScAddr()) {
-                    this.showTool(this.toolChangeType());
-                }
-            } else if (this.scene.selected_objects.length == 1 && !this.scene.selected_objects[0].sc_addr) {
-                if (this.scene.selected_objects[0] instanceof SCg.ModelNode) {
-                    this.showTool(this.toolChangeIdtf());
-                    this.showTool(this.toolChangeType());
-                } else if (this.scene.selected_objects[0] instanceof SCg.ModelEdge) {
-                    this.showTool(this.toolChangeType());
-                } else if (this.scene.selected_objects[0] instanceof SCg.ModelContour) {
-                    this.showTool(this.toolChangeIdtf());
-                } else if (this.scene.selected_objects[0] instanceof SCg.ModelLink) {
-                    this.showTool(this.toolSetContent());
-                }
+    onSelectionChanged: async function () {
+        this.hideTool(this.toolChangeIdtf());
+        this.hideTool(this.toolChangeType());
+        this.hideTool(this.toolSetContent());
+        this.hideTool(this.toolDelete());
+
+        if (this.scene.selected_objects.length > 1) {
+            if (this.scene.isSelectedObjectAllArcsOrAllNodes() && !this.scene.isSelectedObjectAllHaveScAddr()) {
+                this.showTool(this.toolChangeType());
+            }
+        } else if (this.scene.selected_objects.length === 1 && !this.scene.selected_objects[0].sc_addr) {
+            if (this.scene.selected_objects[0] instanceof SCg.ModelNode) {
+                this.showTool(this.toolChangeIdtf());
+                this.showTool(this.toolChangeType());
+            } else if (this.scene.selected_objects[0] instanceof SCg.ModelEdge) {
+                this.showTool(this.toolChangeType());
+            } else if (this.scene.selected_objects[0] instanceof SCg.ModelContour) {
+                this.showTool(this.toolChangeIdtf());
+            } else if (this.scene.selected_objects[0] instanceof SCg.ModelLink) {
+                this.showTool(this.toolSetContent());
+            }
+        } else if (this.scene.selected_objects.length === 1) {
+            if (this.scene.selected_objects[0] instanceof SCg.ModelLink) {
+                this.showTool(this.toolSetContent());
             }
             if (this.scene.selected_objects.length > 0) this.showTool(this.toolDelete());
```

Comment:
This condition should go on level up. We can select multiple objects with Ctrl and delete button should be available as exist